frame.go

```diff
@@ -1060,6 +1068,10 @@ func (f *framer) parseResultMetadata() resultMetadata {
 		meta.pagingState = copyBytes(f.readBytes())
 	}
 
+	if meta.flags&flagMetaDataChanged == flagMetaDataChanged {
+		meta.newMetadataID = copyBytes(f.readBytes())
```

Comment:
we should only check this flag if protoversion is `> 4`. Also the result metadata id is a `[short bytes]` not `[bytes]` (see how it is being parsed in the `PREPARED` response and the v5 spec)

Reply:
Good catch! And this is what tests lacking is leading to...